### PR TITLE
KAFKA-16521: kafka-metadata-quorum describe command changes for KIP-853

### PR DESCRIPTION
**Pull request by ahuang98**
`describe --status` now includes directory id and endpoint information for voter and observers.
`describe --replication` now includes directory id.

It looks like we want to include a `CommittedVoters:` field as well, I think we can include it in a later PR given that it will require some larger changes to plumb that information through.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: tools/src/main/java/org/apache/kafka/tools/MetadataQuorumCommand.java [hunk 1]
```diff
@@ -232,9 +235,60 @@ else if (leader.lastCaughtUpTimestamp().isPresent() && maxLagFollower.lastCaught
...
+
+    private static class Node {
+        private final int id;
+        private final Uuid directoryId;
+        private final List<RaftVoterEndpoint> endpoints;
+
+        private Node(int id, Uuid directoryId, List<RaftVoterEndpoint> endpoints) {
+            this.id = id;
+            this.directoryId = directoryId;
+            this.endpoints = endpoints;
```
Comment (cmccabe) [inline tools/src/main/java/org/apache/kafka/tools/MetadataQuorumCommand.java]: would it be simpler to just do `Objects.requireNonNull` here and skip the null checks in toString? I don't see any way endpoints could be initialized as null here. Same question for `directoryId`.

File: tools/src/main/java/org/apache/kafka/tools/MetadataQuorumCommand.java [hunk 2]
```diff
@@ -232,9 +235,60 @@ else if (leader.lastCaughtUpTimestamp().isPresent() && maxLagFollower.lastCaught
...
+        }
+
+        @Override
+        public String toString() {
+            StringBuilder sb = new StringBuilder();
+            sb.append("{");
+            sb.append("\"id\": ").append(id).append(", ");
```
Comment (cmccabe) [inline tools/src/main/java/org/apache/kafka/tools/MetadataQuorumCommand.java:281]: If it's UUID_ZERO should we print out something special (like "none")?

**Review by cmccabe — APPROVED** [review]

**Comment by jsancio** [comment]
> It looks like we want to include a CommittedVoters: field as well, I think we can include it in a later PR given that it will require some larger changes to plumb that information through.

@ahuang98 I agree. Can you file an issue under https://issues.apache.org/jira/browse/KAFKA-17241 to implement this?

**Review by jsancio — APPROVED** [review]
LGTM

**Comment by ahuang98** [comment]
https://issues.apache.org/jira/browse/KAFKA-17243

**Pull request merged**